src/common/common.h

```diff
@@ -23,10 +24,12 @@ extern "C" {
  * Macro for terminating the program if x is
  * a null pointer.
  */
-#define OQS_EXIT_IF_NULLPTR(x)  \
-    do {                        \
-        if ( (x) == (void*)0 )  \
+#define OQS_EXIT_IF_NULLPTR(x)    \
```

Comment:
Since the `fprintf` error message is OpenSSL specific, could you rename (or move) this to something like `OQS_EXIT_IF_OPENSSL_NULLPTR` or some such?

Reply:
No code other than the _ossl files use this macro. But fine, I'll change this.

Reply:
Changed macro to always state where the problem occurs in https://github.com/open-quantum-safe/liboqs/pull/1454/commits/5f8d02c9c2651b9a5f3eca0364457a5b271a473a. 

I'd suggest looking into improving this in the future: What was the rationale for all these `void` functions? Maybe someone would like to raise an issue documenting what this is about and how it could be improved (at least stating more specifically what/where a problem occurred, if it occurred, such that users of `liboqs` have a better idea where to look for possible program termination causes).

Reply:
I think one reason for all these void functions is that the implementers of all the various PQ implementations we're using were working with SHA/AES APIs that didn't have return values -- they were assumed to just always work.  It would be very intrusive to go back to all of those calls deeply in the source code and change them all to have error handling.  We shouldn't forget about this if and when we start on the APIs for some of the other projects on the production track in the Foundation.

Reply:
> Changed macro to always state where the problem occurs in [5f8d02c](https://github.com/open-quantum-safe/liboqs/commit/5f8d02c9c2651b9a5f3eca0364457a5b271a473a).

I think using `__LINE__` and `__FILE__` macros would've been better to pinpoint the source of the error, but I won't hold up the merge for this since it's not specific to this PR.

Reply:
> I think using `__LINE__` and `__FILE__` macros would've been better to pinpoint the source of the error, but I won't hold up the merge for this since it's not specific to this PR.

Thanks -- this is a good idea. Silly me to not have thought about that earlier. But whatever, I'd vote for merging this now to make headway on #1420. We still have to properly address the old comment https://github.com/open-quantum-safe/liboqs/blob/9f912c957bfe7f4b894aa9661168a310e8dd1a58/src/common/common.h#L40